### ISSUE TITLE
[8.0/Master] Shopping cart does not persist after logout.

### ISSUE THREAD

**Issue by lukebranch** [odoo/odoo#3339]
<h2>Impacted versions:</h2>

<p>8.0 and above</p>

<h2>Steps to reproduce:</h2>

<ul class="task-list">
<li>Login as admin or any other user in runbots for 8.0 or master</li>
<li>Go to website frontend and add products to shopping cart.</li>
<li>Logout and log back in again. </li>
</ul>

<h2>Current behaviour:</h2>

<ul class="task-list">
<li>Items in shopping cart are removed after logout.</li>
</ul>

<h2>Expected behaviour:</h2>

<ul class="task-list">
<li>Shopping cart should persist after logout, and administration options should allow for configuration of shopping cart persistence</li>
</ul>

<h2>Preferred behaviour:</h2>

<h3>In Administration Settings:</h3>
<ul class="task-list">
<li>Enable Persistence (Yes/No)</li>
<li>Persistence Lifetime (seconds)</li>
<li>Enable "Remember Me" (Yes/No)</li>
<li>"Remember Me" Default Value (Yes/No)</li>
<li>Clear Persistence on Log Out (Yes/No)</li>
<li>Persist Shopping Cart (Yes/No)</li>
</ul>

**Comment by watermelonjuice** [comment]
+1

**Comment by jholze** [comment]
@lukebranch something happens here ? Better Cart handling sounds more like an epic :-)

**Comment by lukebranch** [comment]
@jholze ,

Unfortunately I haven't seen any progress on this thread yet. I think SaaS 6 has just been launched in alpha so i'll do some research into whether or not any of these types of features have been integrated or not, otherwise i'll look into building something myself when I can find time.

Currently the cart is being saved as a quotation when the user logs out, so there is some form of persistence, just not the desired behaviour (for me at least). I think it may be possible to modify this process without too much work, however i'll take a look and see if it's already been implemented in one of the odoo-dev branches before getting started.

**Comment by dhongu** [comment]
I made a module to get sale order after login:

```python
class website(models.Model):
    _inherit = 'website' 

    def sale_get_order(self, cr, uid, ids, force_create=False, code=None, update_pricelist=None, context=None):
        sale_order_obj = self.pool['sale.order']
        sale_order_id = request.session.get('sale_order_id')
        if not sale_order_id:
             
            partner = self.pool['res.users'].browse(cr, SUPERUSER_ID, uid, context=context).partner_id
                     
            if partner:
                domain = [('partner_id','=',partner.id), ('state','=','draft')]               
                sale_order_id = sale_order_obj.search(cr, SUPERUSER_ID, domain, limit=1, context=context)
                 
                if sale_order_id:
                    request.session['sale_order_id'] = sale_order_id
            
            
        res = super(website,self).sale_get_order(cr, uid, ids, force_create, code, update_pricelist, context)
        return res
```

**Comment by lukebranch** [comment]
@dhongu ,

I think this feature has already been implemented in Odoo 9.0, however I would suggest creating your module and submitting it to the OCA/website repository for the 8.0 branch, as I believe this feature is definitely missing in Odoo 8.0 and could be useful for many people using 8.0.

**Comment by lapagept** [comment]
@dhongu 
Hello there, should I try your code? Is it still the best  way to get a persistent cart?
